src/lib.rs

```diff
@@ -56,6 +56,24 @@ where
     Err(())
 }
 
+#[cfg(target_env = "gnu")]
+mod glibc {
+    use std::path::Path;
+
+    // Attempt to compile a c program that links to strlcpy from the std
+    // library to determine whether glibc packages it.
+    pub(crate) fn has_strlcpy() -> bool {
+        let path = Path::new(env!("CARGO_MANIFEST_DIR")).join("src/strlcpy.c");
```

Comment:
If this pattern occurs more often, the CMake macro generates this file on the fly and doesn't even make any attempt to reproduce the signature faithfully.

Reply:
Yeah. If in the future, if we need to do anything more complex, it would make sense to dynamically generate a file containing all the required links. For now I think its fine.